src/pts/pyspark/common/ontology.py

```diff
@@ -0,0 +1,88 @@
+"""Map diseases using OnToma."""
+
+from loguru import logger
+from ontoma import OnToma
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.functions import array, array_union, coalesce, col, explode_outer, lit
+
+
+def add_efo_mapping(
+    spark: SparkSession,
+    evidence_df: DataFrame,
+    label_col_name: str,
+    disease_label_lut_path: str,
+    id_col_name: str | None = None,
+    disease_id_lut_path: str | None = None
+) -> DataFrame:
+    """Adds a column containing EFO mappings to the provided dataframe.
+
+    Given a dataframe containing evidence with columns containing disease label and id information,
+    this function returns the dataframe with the additional column diseaseFromSourceMappedId containing
+    the corresponding EFO mappings.
+    If there are multiple mappings for an evidence, the results are exploded accordingly.
+
+    Args:
+        spark (SparkSession): Spark session.
+        evidence_df (DataFrame): Dataframe containing evidence.
+        label_col_name (str): Name of the column containing the disease label.
+        disease_label_lut_path (str): Path to the disease label lookup table.
+        id_col_name (str | None): Name of the column containing the disease id.
+        disease_id_lut_path (str | None): Path to the disease id lookup table.
+
+    Returns:
+        DataFrame: Dataframe with an additional column diseaseFromSourceMappedId containing the EFO mappings.
+    """
+    logger.info(f'Load disease label lookup table from {disease_label_lut_path}.')
+    disease_label_lut = OnToma(spark=spark, cache_dir=disease_label_lut_path)
+    label_mapped_df = disease_label_lut.map_entities(
+        evidence_df, 'labelMappedResult', label_col_name, 'label', type_col=lit('DS')
+    )
+
+    if id_col_name and disease_id_lut_path:
+
+        logger.info(f'Load disease id lookup table from {disease_id_lut_path}.')
+        disease_id_lut = OnToma(spark=spark, cache_dir=disease_id_lut_path)
+        label_id_mapped_df = disease_id_lut.map_entities(
+            label_mapped_df, 'idMappedResult', id_col_name, 'id', type_col=lit('DS')
+        )
+
+        result_df = _combine_result_columns(label_id_mapped_df, 'labelMappedResult', 'idMappedResult')
+        logger.info(
+            f'Mapped diseases from {evidence_df.count()} evidence using the'
```

Comment:
I suggest not to extract metrics during runtime, this will trigger spark execution
```suggestion
```